runtime/Cpp/runtime/src/atn/ParserATNSimulator.cpp

```diff
@@ -697,24 +697,26 @@ std::vector<Ref<SemanticContext>> ParserATNSimulator::getPredsForAmbigAlts(const
 
 std::vector<dfa::DFAState::PredPrediction *> ParserATNSimulator::getPredicatePredictions(const antlrcpp::BitSet &ambigAlts,
   std::vector<Ref<SemanticContext>> altToPred) {
-  std::vector<dfa::DFAState::PredPrediction*> pairs;
   bool containsPredicate = false;
   for (size_t i = 1; i < altToPred.size(); i++) {
-    Ref<SemanticContext> pred = altToPred[i];
+    const Ref<SemanticContext>& pred = altToPred[i];
 
     // unpredicted is indicated by SemanticContext.NONE
     assert(pred != nullptr);
 
-    if (ambigAlts.test(i)) {
-      pairs.push_back(new dfa::DFAState::PredPrediction(pred, (int)i)); /* mem-check: managed by the DFAState it will be assigned to after return */
-    }
     if (pred != SemanticContext::NONE) {
       containsPredicate = true;
+      break;
```

Comment:
I don't think using `break` here is a good idea, since it is supposed to check all `pred` is not `nullptr`.

Reply:
It just needs to find one predicate that is non NONE.

Reply:
@mike-lischke I mean there is still an assertion statement `assert(pred != nullptr);` above in the loop and the `break` makes the check not to cover all `pred`, in this commit. Whatever, the "c++-ish" solution you mentioned is good with it.

Reply:
The assertion is probably not needed anymore. But it doesn't hurt either.

The first loop is fine because all it needs to check is that at least one non-NONE context is in the list. So it can break on the first one it finds.

Reply:
@mike-lischke I know that, but I think you didn't get my point. I mean, if a `pred` is not NONE and `break` here in this case, the assertion will not be executed for the other `pred`s afterwards.

Reply:
I have stuck the assert in the second loop. But in general relying on asserts for correctness is very bad style, so the function flow should not be altered to trigger an assert (I think).